nixos/modules/services/networking/thelounge.nix

```diff
@@ -0,0 +1,85 @@
+{ pkgs, lib, config, ... }:
+
+with lib;
+
+let
+  cfg = config.services.thelounge;
+  configJsData = "module.exports = " + builtins.toJSON (
+    { private = cfg.private; port = cfg.port; } // cfg.extraConfig
+  );
+in {
+  options.services.thelounge = {
+    enable = mkEnableOption "The Lounge web IRC client";
+
+    home = mkOption {
+      type = types.path;
+      default = "/var/lib/thelounge";
```

Comment:
I don't think there is a strong need for this to be configurable.
This can be manged using systemd's  `StateDirectory=` that take also care of changing the permission on service startup.